src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveWorkflow.cs

```diff
@@ -174,7 +170,18 @@ public sealed class RInteractiveWorkflow : IRInteractiveWorkflow {
             if (!RSessions.HasBroker) {
                 var connectedToBroker = await Connections.TryConnectToPreviouslyUsedAsync();
                 if (!connectedToBroker) {
-                    Connections.GetOrCreateVisualComponent().Container.Show(focus:false, immediate:false);
+                    var showConnectionsWindow = Connections.RecentConnections.Any();
+                    if (!showConnectionsWindow){
+                        var message = Resources.NoLocalR.FormatInvariant(Environment.NewLine + Environment.NewLine, Environment.NewLine);
+                        showConnectionsWindow = Shell.ShowMessage(message, MessageButtons.YesNo) == MessageButtons.Yes;
+                    }
+
+                    if (!showConnectionsWindow) {
+                        var installer = Shell.ExportProvider.GetExportedValue<IMicrosoftRClientInstaller>();
+                        installer.LaunchRClientSetup(Shell);
+                    } else {
+                        Connections.GetOrCreateVisualComponent().Container.Show(focus: false, immediate: false);
```

Comment:
I guess it may be OK if user uninstalled R they used to use, but then we should pick up another one automatically.